src/Gui/MainWindow.cpp

```diff
@@ -2509,97 +2512,132 @@ QMdiArea *MainWindow::getMdiArea() const
     return d->mdiArea;
 }
 
-// ----------------------------------------------------------
-
-StatusBarObserver::StatusBarObserver()
-  : WindowParameter("OutputWindow")
+void MainWindow::setWindowTitle(const QString& string)
 {
-    msg = QString::fromLatin1("#statusBar{color: #000000}"); // black
-    wrn = QString::fromLatin1("#statusBar{color: #ffaa00}"); // orange
-    err = QString::fromLatin1("#statusBar{color: #ff0000}"); // red
-    Base::Console().AttachObserver(this);
-    getWindowParameter()->Attach(this);
-    getWindowParameter()->NotifyAll();
-}
+    QString title;
+    QString appname =
+        QString(static_cast<QApplication*>(QCoreApplication::instance())->applicationName());
```

Comment:
This is awkward. applicationName() is a static method of QCoreApplication.

Reply:
This function was copied from their previous location, I did no change here.